designer/server/src/main/scala/pl/touk/nussknacker/ui/api/typingDto.scala

```diff
@@ -0,0 +1,363 @@
+          List(
+            SProductField[String, String](FieldName("display"), Schema.string, display => Some(display)),
+            SProductField[String, String](FieldName("type"), Schema.string, typ => Some(typ)),
+            SProductField[String, Dict](FieldName("dict"), dictSchema, dict => None),
+          )
+        ),
+        Some(SName("TypedDictDto"))
+      )
+        .title("TypedDictDto")
+        .as
+    }
+
+    def apply(typedDict: TypedDict)(implicit modelData: ModelData): TypedDictDto = {
+      new TypedDictDto(typedDict.dictId, SingleTypingResultDto.apply(typedDict.valueType))
+    }
+
+  }
+
+  @derive(schema)
+  sealed trait TypedObjectWithDataDto extends SingleTypingResultDto {
+    def underlying: SingleTypingResultDto
+    def data: Any
+    override def objType: TypedClassDto = underlying.objType
+
+    implicit val typedObjectWithDataDtoSchema: Schema[TypedObjectWithDataDto] =
+      Schema.derived.hidden(true)
+  }
+
+  case class TypedTaggedValueDto(underlying: SingleTypingResultDto, tag: String) extends TypedObjectWithDataDto {
+    override def data: String = tag
+
+    override def valueOpt: Option[Any] = underlying.valueOpt
+
+    override def withoutValue: TypedTaggedValueDto = TypedTaggedValueDto(underlying.withoutValue, tag)
+
+    override def display: String = s"${underlying.display} @ $tag"
+
+  }
+
+  object TypedTaggedValueDto {
+
+    implicit lazy val typedTaggedSchema: Schema[TypedTaggedValueDto] = {
+      implicit val typingResultSchema: Schema[TypingResultDto] = Schema.derived[TypingResultDto]
+      Schema(
+        SchemaType.SProduct(
+          List(
+            SProductField[String, String](FieldName("tag"), Schema.string, tag => Some(tag)),
+            SProductField[String, String](FieldName("display"), Schema.string, display => Some(display)),
+            SProductField[String, String](FieldName("type"), Schema.string, typ => Some(typ)),
+            SProductField[String, String](FieldName("refClazzName"), Schema.string, refClazzName => Some(refClazzName)),
+            SProductField[String, List[TypingResultDto]](
+              FieldName("params"),
+              Schema.schemaForArray[TypingResultDto].as,
+              sth => Some(List(UnknownDto))
+            )
+          )
+        ),
+        Some(SName("TypedTaggedValueDto"))
+      )
+        .title("TypedTaggedValueDto")
+        .as
+    }
+
+    def apply(typedTaggedValue: TypedTaggedValue)(implicit modelData: ModelData): TypedTaggedValueDto = {
+      TypedTaggedValueDto(SingleTypingResultDto.apply(typedTaggedValue.underlying), typedTaggedValue.tag)
+    }
+
+  }
+
+  case class TypedObjectWithValueDto private[typingDto] (
+      underlying: TypedClassDto,
+      value: Any,
+      waitRefClazzName: String = "",
+      waitValue: Json = Json.fromString("")
+  ) extends TypedObjectWithDataDto {
+    val maxDataDisplaySize: Int         = 15
+    val maxDataDisplaySizeWithDots: Int = maxDataDisplaySize - "...".length
```

Comment:
I understand it as "max amount of data displayed when dots are needed" (as it's longer then 15) but this part was also copied from `typing.scala` so that's only my thoughts of what it means